pkg/operators/clustermanager/controllers/statuscontroller/clustermanager_status_controller.go

```diff
@@ -72,29 +73,29 @@ func (s *clusterManagerStatusController) sync(ctx context.Context, controllerCon
 	}
 
 	errs := []error{}
-	if err := s.updateStatusOfRegistration(ctx, clusterManager.Name); err != nil {
+	if err := s.updateStatusOfRegistration(ctx, clusterManager.Name, clusterManager.Spec.DeployOption.Mode); err != nil {
 		errs = append(errs, err)
 	}
 
-	if err := s.updateStatusOfPlacement(ctx, clusterManager.Name); err != nil {
+	if err := s.updateStatusOfPlacement(ctx, clusterManager.Name, clusterManager.Spec.DeployOption.Mode); err != nil {
 		errs = append(errs, err)
 	}
 
 	return operatorhelpers.NewMultiLineAggregate(errs)
 }
 
 // updateStatusOfRegistration checks registration deployment status and updates condition of clustermanager
-func (s *clusterManagerStatusController) updateStatusOfRegistration(ctx context.Context, clusterManagerName string) error {
+func (s *clusterManagerStatusController) updateStatusOfRegistration(ctx context.Context, clusterManagerName string, mode v1.InstallMode) error {
 	// Check registration deployment status
 	registrationDeploymentName := fmt.Sprintf("%s-registration-controller", clusterManagerName)
-	registrationDeployment, err := s.deploymentLister.Deployments(helpers.ClusterManagerNamespace).Get(registrationDeploymentName)
+	registrationDeployment, err := s.deploymentLister.Deployments(helpers.ClusterManagerNamespace(clusterManagerName, mode)).Get(registrationDeploymentName)
 	if err != nil {
 		_, _, err := helpers.UpdateClusterManagerStatus(ctx, s.clusterManagerClient, clusterManagerName,
 			helpers.UpdateClusterManagerConditionFn(metav1.Condition{
 				Type:    registrationDegraded,
 				Status:  metav1.ConditionTrue,
 				Reason:  "GetRegistrationDeploymentFailed",
-				Message: fmt.Sprintf("Failed to get registration deployment %q %q: %v", helpers.ClusterManagerNamespace, registrationDeploymentName, err),
+				Message: fmt.Sprintf("Failed to get registration deployment %q %q: %v", helpers.ClusterManagerNamespace(clusterManagerName, mode), registrationDeploymentName, err),
```

Comment:
can we get installNamespace at the top rather than call this every time

Reply:
Sure we can.

Reply:
Solved.